modules/python/python-dest.c

```diff
@@ -185,10 +185,56 @@ _py_invoke_close(PythonDestDriver *self)
   _dd_py_invoke_void_method_by_name(self, "close");
 }
 
-static gboolean
+static worker_insert_result_t
+_as_int(PyObject *obj)
+{
+  int result = pyobject_as_int(obj);
+  if (result == -1 && PyErr_Occurred())
+    {
+      gchar buf[256];
+      msg_error("Error converting PyObject to int. Dropping message",
+                evt_tag_str("exception", _py_format_exception_text(buf, sizeof(buf))));
+      _py_finish_exception_handling();
+      return WORKER_INSERT_RESULT_DROP;
+    }
+
+  if (result < 0 || result >= WORKER_INSERT_RESULT_MAX)
```

Comment:
In case of the other language binding (java) I did not saw such safe guarding. Would it be possible to keep them in sync ?

Is it right to drop the message ?

Reply:
Good idea, I will push a patch on the top of https://github.com/balabit/syslog-ng/pull/2478., just let's agree on the behaviour first.

I think the dropping is preferable here.
From development point of view, the behaviour does not really matter, the interesting question is when the condition hits a production environment. When this condition occurs, it is already a very bad situation, meaning that the driver implementation is faulty. It should not be that faulty that the condition is triggered for every messages (talking about production code), the problem should be message specific. When talking about message specific, it is probably not worth retrying the same message, because the same input will probably trigger the same condition.

With that said, I think the best is to drop the message so that other messages could be handled without delay.
The second best option would be disconnect, which will suspend the driver (probably infinitely as the same message will be retried over and over), when the messages will accumulate into the queue. This would minimize message loss to some extent, but depending on how quickly the problem is detected and addressed by the operators, this will still result in either message loss in syslog-ng (without flow control), or congestion at some other part of the operator network. And causes the largest possible delay in the messages depending how quickly can the message queue be salvaged.
The least preferable option is retry mechanism, which would be the most optimistic version, but I think if a code path fails on a message, it will fail again later.

What do you think?

Reply:
Okok, I can align align to that.

Also, just occurred to me that any uncatched exception in the send method is very similar scenario to this one: and we return `WORKER_INSERT_RESULT_ERROR` in such case. So using `WORKER_INSERT_RESULT_ERROR` here is the consistent choice.